bucket.tf

```diff
@@ -37,13 +27,19 @@ resource "aws_s3_bucket" "web" {
   acl           = "${var.bucket_acl}"
   force_destroy = "${var.bucket_force_destroy}"
 
-  website = {
-    index_document = "${var.default_root_object}"
-  }
-
   tags = "${merge(map("Name", format("%s", "Bucket for CloudFront ${var.environment}")), map("Environment", format("%s", var.environment)), var.tags)}"
 
   versioning {
     enabled = "${var.bucket_versioning}"
   }
+
```

Comment:
Why do you add lifecycle managment to the bucket?

Reply:
I do not think versioning is required, but it is already there.
Since versioning is possible to enable, not having a lifecycle rule will result in old versions staying indefinitely. That means the bucket size will grow and there is no way to manage it automatically.